test/dotnet-new3.UnitTests/AllProjectsWork.cs

```diff
@@ -1,78 +1,75 @@
-//        [InlineData("mvc_cs-10", "mvc")]
-//        [InlineData("mvc_fs-10", "mvc", "-lang", "F#")]
-//        [InlineData("api_cs-10", "api")]
-//        [InlineData("emptyweb_cs-11", "web", "-f", "1.1")]
-//        [InlineData("mvc_cs-11", "mvc", "-f", "1.1")]
-//        [InlineData("mvc_fs-11", "mvc", "-lang", "F#", "-f", "1.1")]
-//        [InlineData("api_cs-11", "api", "-f", "1.1")]
-//        [InlineData("console_cs-10", "console")]
-//        [InlineData("library_cs-10", "library")]
-//        public void AllWebProjectsRestoreAndBuild(string testName, params string[] args)
-//        {
-//            string workingDir = GetWorkingDirectoryName(testName);
-//            Directory.CreateDirectory(workingDir);
+        public AllProjectsWork(SharedHomeDirectory sharedHome, ITestOutputHelper log)
+        {
+            this.sharedHome = sharedHome;
+            this.log = log;
+            sharedHome.InstallPackage("Microsoft.DotNet.Web.ProjectTemplates.5.0");
+            sharedHome.InstallPackage("Microsoft.DotNet.Web.ProjectTemplates.3.1");
+        }
 
-//            Program.Main(new[] { "--debug:reinit" });
-            
-//            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable("DN3")))
-//            {
-//                string relTo = new Uri(typeof(Program).GetTypeInfo().Assembly.CodeBase, UriKind.Absolute).LocalPath;
-//                relTo = Path.GetDirectoryName(relTo);
-//                relTo = Path.Combine(relTo, @"..\..\..\..\..\dev");
-//                Environment.SetEnvironmentVariable("DN3", relTo);
-//            }
+        [Theory]
+        [InlineData("emptyweb_cs-50",  "web")]
+        [InlineData("mvc_cs-50",  "mvc")]
+        [InlineData("mvc_fs-50",  "mvc", "-lang", "F#")]
+        [InlineData("api_cs-50",  "api")]
+        [InlineData("emptyweb_cs-31",  "web", "-f", "netcoreapp3.1")]
+        [InlineData("mvc_cs-31",  "mvc", "-f", "netcoreapp3.1")]
+        [InlineData("mvc_fs-31",  "mvc", "-lang", "F#", "-f", "netcoreapp3.1")]
+        [InlineData("api_cs-31",  "api", "-f", "netcoreapp3.1")]
+        [InlineData("console_cs-31", "console", "-f", "netcoreapp3.1")]
+        [InlineData("library_cs-50", "classlib", "-f", "net5.0")]
+        public void AllWebProjectsRestoreAndBuild(string testName, params string[] args)
+        {
+            string workingDir = Helpers.CreateTemporaryFolder(testName);
 
-//            CommandResult result = Command.Create("dotnet-new3", args, outputPath: Environment.GetEnvironmentVariable("DN3"))
-//                .WorkingDirectory(workingDir)
-//                .CaptureStdErr()
-//                .CaptureStdOut()
-//                .Execute();
-//            Assert.Equal(0, result.ExitCode);
+            new DotnetNewCommand(log, args)
+                .WithWorkingDirectory(workingDir)
+                .WithEnvironmentVariable(sharedHome.HomeVariable, sharedHome.HomeDirectory)
+                .Execute()
+                .Should()
```

Comment:
In unit test, it is sometimes recommended to keep AAA pattern - https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80
In this particular cases I believe it would be better to split between execute and asserting like:
```C#
var resultOfSomething = new DotnetNewCommand(log, args)....Execute();
resultOfSomething.Shoud().BeSomething();
```
Not only it makes it more readable, to me, but it also allows easier debugging, when needed.

Reply:
I kinda agree, but I want to stay close to dotnet/sdk style...